include/mumble/Base64.hpp

```diff
@@ -3,31 +3,175 @@
 // that can be found in the LICENSE file at the root of the
 // Mumble source tree or at <https://www.mumble.info/LICENSE>.
 
+// Inspired by https://renenyffenegger.ch/notes/development/Base64/Encoding-and-decoding-base-64-with-cpp.
+
 #ifndef MUMBLE_BASE64_HPP
 #define MUMBLE_BASE64_HPP
 
-#include "Macros.hpp"
-#include "NonCopyable.hpp"
 #include "Types.hpp"
 
-#include <memory>
+#include <array>
+#include <cmath>
+#include <limits>
+#include <stdexcept>
+
+#include <gsl/span>
 
 namespace mumble {
-class MUMBLE_EXPORT Base64 : NonCopyable {
-public:
-	class P;
+namespace base64 {
+	using StrView      = gsl::span< char >;
+	using StrViewConst = gsl::span< const char >;
+
+	static constexpr char PAD_CHAR = '=';
+
+	static constexpr std::byte operator+(const std::byte lhs, const std::byte rhs) {
+		const auto ret = std::to_integer< uint8_t >(lhs) + std::to_integer< uint8_t >(rhs);
+		return std::byte(ret);
+	}
+
+	// The standard doesn't overload this operator with both operands as std::byte...
+	static constexpr std::byte operator<<(const std::byte lhs, const std::byte rhs) {
+		return lhs << std::to_integer< uint8_t >(rhs);
+	}
+
```

Comment:
```suggestion
```
This is not needed if you apply my suggestions from above

Reply:
It's required, otherwise we end up with compile warnings due to `int` being the implicit data type.